fea-rs/src/parse/source.rs

```diff
@@ -64,32 +165,12 @@ impl FileId {
 }
 
 impl Source {
-    /// Attempts to generate a `Source` from the contents of the provided path.
-    pub fn from_path(path: impl Into<PathBuf>) -> Result<Self, SourceLoadError> {
-        let path = path.into();
-        let contents = std::fs::read_to_string(&path).map_err(|cause| SourceLoadError {
-            path: path.clone(),
-            cause,
-        })?;
-        let line_offsets = line_offsets(&contents);
-        Ok(Source {
-            path: Some(path),
-            id: FileId::next(),
-            contents: contents.into(),
-            line_offsets,
-        })
-    }
-
-    /// Create a source from a string.
-    ///
-    /// This is useful for things like testing.
-    pub fn from_text(contents: impl Into<Arc<str>>) -> Source {
-        let contents = contents.into();
+    pub(crate) fn new(path: impl Into<OsString>, contents: Arc<str>) -> Self {
```

Comment:
Perhaps just due to Rust noobness both Arc and str threw me here. Perhaps at the very least a comment is in order? - I think I'd have expected just &str because afaik we don't thread.

Reply:
Fair enough, this is a pattern we've used in a bunch of projects previously, but it's reasonable that you aren't familiar with it. :)